### DISCUSSION TITLE
Postfix default behavior

### DISCUSSION THREAD

**Discussion by MunifTanjim** [L3MON4D3/LuaSnip#897] [Q&A] [answered]
Shouldn't postfix by default require that there are text before it? For example:

```lua
local increment = postfix({
  trig = "++",
  hidden = true,
  condition = has_word_before,
}, {
  f(function(_, parent)
    return string.format("%s = %s + 1", parent.snippet.env.POSTFIX_MATCH, parent.snippet.env.POSTFIX_MATCH)
  end, {}),
})
```

I'm doing this, using `condition` to enforce that there are non-whitespace characters before `++` trigger.

If there are whitespace characters before the trigger, it's not really _**postfix**_ anymore, right?

**Comment by L3MON4D3** [comment]
Mhmmm I guess we could introduce that requirement.. is there any benefit beside this being conceptually cleaner?

**Reply by MunifTanjim** [reply]
> is there any benefit beside this being conceptually cleaner?

The current behavior feels a bit buggy.

For example,

```
  ++
```

expanding this generates

```
   =  + 1
```

The main feature of `postfix` is the `parent.snippet.env.POSTFIX_MATCH` variable, if that's empty expanding the snippet isn't useful.

**Reply by L3MON4D3** [reply, marked as answer]
Yeeeeaaaah okay, I think that's one of those things that mainly happens while testing, and one then never notices while actually doing something else :D
But it would be nice to not display postfix-snippets in eg. `nvim-cmp` if they don't actually match at the moment.
And in any case, including the condition/show_condition should not hurt (but we should include an option to turn this behaviour off, just to be safe)
